Doc/tools/extensions/pyspecific.py

```diff
@@ -459,6 +460,10 @@ def run(self):
         content = issue_re.sub(r'`bpo-\1 <https://bugs.python.org/'
                                r'issue?@action=redirect&bpo=\1>`__',
                                content)
+        # Fallback handling for the GitHub issue
+        content = gh_issue_re.sub(r'`gh-issue-\1 <https://github.com/'
+                                  r'python/cpython/issues/\1>`__',
+                                  content)
```

Comment:
What is `content` here?
AFAIK `gh-issue` is only used in the filename of the Misc/NEWS entries and as a Sphinx directive (`.. gh-issue:`) in the NEWS files' content. otherwise I've seen GitHub issues referenced as either `GH-NNNN`/`gh-NNNN`, or `` :gh:`...` ``.

Reply:
Looks like it comes from https://github.com/python/core-workflow/blob/master/blurb/blurb.py#L1116.  I wonder if this should be changed to just `gh-NNNN: ` in the output.

Reply:
I created this:
* python/core-workflow#451

Reply:
> What is content here?

@ezio-melotti 

It's a raw string of changelog: https://docs.python.org/3.11/whatsnew/changelog.html#changelog,
We replaced following rawstring from `as-is` to `to-be`.

### AS-IS
`````

 content[:2000]
"+++++++++++\nPython News\n+++++++++++\n\nWhat's New in Python 3.11.0 beta 1?\n===================================\n\n*Release date: 2022-05-06*\n\nSecurity\n--------\n\n- gh-issue-57684: Add the :option:`-P` command line option and the\n  :envvar:`PYTHONSAFEPATH` environment variable to not prepend a potentially\n  unsafe path to :data:`sys.path`. Patch by Victor Stinner.\n\nCore and Builtins\n-----------------\n\n- gh-issue-89519: Chaining classmethod descriptors (introduced in bpo-19072)\n  is deprecated. It can no longer be used to wrap other descriptors such as\n  property().  The core design of this feature was flawed, and it caused a\n  number of downstream problems.\n\n- gh-issue-92345: ``pymain_run_python()`` now imports ``readline`` and\n  ``rlcompleter`` before sys.path is extended to include the current working\n  directory of an interactive interpreter. Non-interactive interpreters are\n  not affected.\n\n- bpo-43857: Improve the :exc:`AttributeError` message when deleting a\n  missing attribute. Patch by Géry Ogam.\n\n- gh-issue-92245: Make sure that PEP 523 is respected in all cases. In\n  3.11a7, specialization may have prevented Python-to-Python calls\n  respecting PEP 523.\n\n- gh-issue-92203: Add a closure keyword-only parameter to exec(). It can\n  only be specified when exec-ing a code object that uses free variables.\n  When specified, it must be a tuple, with exactly the number of cell\n  variables referenced by the code object. closure has a default value of\n  None, and it must be None if the code object doesn't refer to any free\n  variables.\n\n- gh-issue-91173: Disable frozen modules in debug builds. Patch by Kumar\n  Aditya.\n\n- gh-issue-92114: Improve error message when subscript a type with\n  ``__class_getitem__`` set to ``None``.\n\n- gh-issue-92112: Fix crash triggered by an evil custom ``mro()`` on a\n  metaclass.\n\n- gh-issue-92063: The ``PRECALL_METHOD_DESCRIPTOR_FAST_WITH_KEYWORDS``\n  instruction now ensures methods are called only on objects of the correct\n  type.\n\n- gh-issue-9"
`````

###TO-BE
``````

(Pdb) content[:2000]
"+++++++++++\nPython News\n+++++++++++\n\nWhat's New in Python 3.11.0 beta 1?\n===================================\n\n*Release date: 2022-05-06*\n\nSecurity\n--------\n\n- `gh-issue-57684 <https://github.com/python/cpython/issues/57684>`__: Add the :option:`-P` command line option and the\n  :envvar:`PYTHONSAFEPATH` environment variable to not prepend a potentially\n  unsafe path to :data:`sys.path`. Patch by Victor Stinner.\n\nCore and Builtins\n-----------------\n\n- `gh-issue-89519 <https://github.com/python/cpython/issues/89519>`__: Chaining classmethod descriptors (introduced in `bpo-19072 <https://bugs.python.org/issue?@action=redirect&bpo=19072>`__)\n  is deprecated. It can no longer be used to wrap other descriptors such as\n  property().  The core design of this feature was flawed, and it caused a\n  number of downstream problems.\n\n- `gh-issue-92345 <https://github.com/python/cpython/issues/92345>`__: ``pymain_run_python()`` now imports ``readline`` and\n  ``rlcompleter`` before sys.path is extended to include the current working\n  directory of an interactive interpreter. Non-interactive interpreters are\n  not affected.\n\n- `bpo-43857 <https://bugs.python.org/issue?@action=redirect&bpo=43857>`__: Improve the :exc:`AttributeError` message when deleting a\n  missing attribute. Patch by Géry Ogam.\n\n- `gh-issue-92245 <https://github.com/python/cpython/issues/92245>`__: Make sure that PEP 523 is respected in all cases. In\n  3.11a7, specialization may have prevented Python-to-Python calls\n  respecting PEP 523.\n\n- `gh-issue-92203 <https://github.com/python/cpython/issues/92203>`__: Add a closure keyword-only parameter to exec(). It can\n  only be specified when exec-ing a code object that uses free variables.\n  When specified, it must be a tuple, with exactly the number of cell\n  variables referenced by the code object. closure has a default value of\n  None, and it must be None if the code object doesn't refer to any free\n  variables.\n\n- `gh-issue-91173 <https://github.com/python/cpython/is"
``````

Reply:
We discussed on Discord and decided that replacing bpo/gh ids with the corresponding roles (`` :issue:`...` `` and `` :gh:`...` ``) instead of hardcoding the link would be better.  By doing this not only we simplified the code and avoided duplication, but we were also able to spot some mistakes thanks to some extra logic baked into the roles.